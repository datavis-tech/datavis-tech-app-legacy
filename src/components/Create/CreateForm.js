import React, { Component } from 'react'
import TitleFormGroup from './TitleFormGroup'
import DescriptionFormGroup from './DescriptionFormGroup'
import ContentUploadFormGroup from './ContentUploadFormGroup'

export default class CreateForm extends Component {
  constructor() {
    super()

    this.state = {

      // The text from the title input field.
      title: "",

      // A flag to trigger validation feedback
      // when the form was submitted with no title present.
      titleDanger: false,

      // The text from the description input field.
      description: "",

      // A flag to signify that the document is being created.
      creating: false
    }

    this.handleChange = this.handleChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
    this.setTitleInput = this.setTitleInput.bind(this)
  }

  handleChange({ target: {id, value}}) {
    this.setState({ [id]: value })

    if(id === 'title' && value){
      this.setState({ titleDanger: false })
    }
  }

  handleSubmit(event) {

    const {
      state: { title, description },
      props: { createDocument, router }
    } = this

    event.preventDefault()

    if(!title){
      this.setState({ titleDanger: true })
      this.focusTitle()
    } else {
      this.setState({ creating: true })
      createDocument(title, description)
        .then((id) => router.push("/" + id))
    }
  }

  setTitleInput(input){
    this.titleInput = input
  }

  focusTitle() {
    if(this.titleInput){
      this.titleInput.focus()
    }
  }
  
  componentDidMount() {
    this.focusTitle()
  }

  render() {
    const titleDanger = this.state.titleDanger
    return (
      <div className="container m-t-1">
        <h1>Create a Document</h1>
        <form onSubmit={this.handleSubmit}>

          <TitleFormGroup
            titleDanger={titleDanger}
            handleChange={this.handleChange}
            setTitleInput={this.setTitleInput}
          />

          <DescriptionFormGroup handleChange={this.handleChange} />

          <ContentUploadFormGroup />

          <div className="form-group row">
            <div className="offset-sm-2 col-sm-10">
              <button
                type="submit"
                className="btn btn-primary"
                disabled={this.state.creating}
              >
                Create
              </button>
              {(() => {
                if(this.state.creating){
                  return (
                    <div className="m-t-1">
                      Creating document...
                    </div>
                  )
                }
              })()}
            </div>
          </div>
        </form>
      </div>
    )
  }
}
