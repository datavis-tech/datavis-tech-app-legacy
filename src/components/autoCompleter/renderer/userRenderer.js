import { Label } from 'semantic-ui-react'

export default ({ title }) => <Label content={title} />
