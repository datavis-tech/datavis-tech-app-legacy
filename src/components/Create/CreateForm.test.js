import React from 'react'
import renderer from 'react-test-renderer'
import CreateForm from './CreateForm'

it('renders', () => {
  const tree = renderer.create(<CreateForm/>).toJSON();
  expect(tree).toMatchSnapshot();
})

it('validates against empty title and provides feedback', () => {
  const render = renderer.create(<CreateForm/>)
  const tree = render.toJSON()
  const form = tree.children[1]

  // Simulate submitting the form.
  let defaultPrevented = false
  form.props.onSubmit({
    preventDefault: () => defaultPrevented = true
  })
  expect(defaultPrevented).toBe(true)

  // This snapshot contains the title danger feedback.
  expect(render.toJSON()).toMatchSnapshot()
})

////it('calls createDocument when title is present', (done) => {
////  const div = document.createElement('div')
////  let createDocumentCalled = false
////  ReactDOM.render(<CreateForm createDocument={ ({title}) => {
////    createDocumentCalled = true
////  }}/>, div)
////  const title = div.querySelector('#title')
////  title.value = 'My Title'
////  title.dispatchEvent(new Event('change', {id:'title', value:'foo'}))
////  setTimeout(() =>{
////    console.log("foo")
////    done()
////  }, 1000)
////
////  var evt = document.createEvent("HTMLEvents");
////  evt.initEvent("change", false, true);
////  title.dispatchEvent(evt);
////
////  //div.querySelector('button').dispatchEvent(new Event('click'))
////  //expect(div.querySelectorAll('.title-feedback').length).toBe(0)
////  //expect(createDocumentCalled).toBe(true)
////})
