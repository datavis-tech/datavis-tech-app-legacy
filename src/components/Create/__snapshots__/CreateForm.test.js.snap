exports[`test renders 1`] = `
<div
  className="container m-t-1">
  <h1>
    Create a Document
  </h1>
  <form
    onSubmit={[Function bound handleSubmit]}>
    <div
      className="form-group row">
      <label
        className="col-sm-2 col-form-label"
        htmlFor="title">
        Title
      </label>
      <div
        className="col-sm-10">
        <input
          className="form-control"
          id="title"
          onChange={[Function bound handleChange]}
          type="text" />
      </div>
    </div>
    <div
      className="form-group row">
      <label
        className="col-sm-2 col-form-label"
        htmlFor="description">
        Description
      </label>
      <div
        className="col-sm-10">
        <textarea
          className="form-control"
          id="description"
          onChange={[Function bound handleChange]}
          rows="3" />
      </div>
    </div>
    <div
      className="form-group row">
      <label
        className="col-sm-2 col-form-label"
        htmlFor="file">
        <span>
          Content
        </span>
      </label>
      <div
        className="col-sm-10">
        <label
          className="custom-file">
          <input
            className="custom-file-input"
            id="file"
            onChange={[Function onChange]}
            type="file" />
          <span
            className="custom-file-control"
            data-content-value="Choose file..." />
        </label>
      </div>
    </div>
    <div
      className="form-group row">
      <div
        className="offset-sm-2 col-sm-10">
        <button
          className="btn btn-primary"
          disabled={false}
          type="submit">
          Create
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`test validates against empty title and provides feedback 1`] = `
<div
  className="container m-t-1">
  <h1>
    Create a Document
  </h1>
  <form
    onSubmit={[Function bound handleSubmit]}>
    <div
      className="form-group row has-danger">
      <label
        className="col-sm-2 col-form-label"
        htmlFor="title">
        Title
      </label>
      <div
        className="col-sm-10">
        <input
          className="form-control form-control-danger"
          id="title"
          onChange={[Function bound handleChange]}
          type="text" />
        <div
          className="form-control-feedback title-feedback">
          A title is required.
        </div>
      </div>
    </div>
    <div
      className="form-group row">
      <label
        className="col-sm-2 col-form-label"
        htmlFor="description">
        Description
      </label>
      <div
        className="col-sm-10">
        <textarea
          className="form-control"
          id="description"
          onChange={[Function bound handleChange]}
          rows="3" />
      </div>
    </div>
    <div
      className="form-group row">
      <label
        className="col-sm-2 col-form-label"
        htmlFor="file">
        <span>
          Content
        </span>
      </label>
      <div
        className="col-sm-10">
        <label
          className="custom-file">
          <input
            className="custom-file-input"
            id="file"
            onChange={[Function onChange]}
            type="file" />
          <span
            className="custom-file-control"
            data-content-value="Choose file..." />
        </label>
      </div>
    </div>
    <div
      className="form-group row">
      <div
        className="offset-sm-2 col-sm-10">
        <button
          className="btn btn-primary"
          disabled={false}
          type="submit">
          Create
        </button>
      </div>
    </div>
  </form>
</div>
`;
